services/rsources/http/http.go

```diff
@@ -89,6 +127,31 @@ func (h *handler) getStatus(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *handler) deleteJobStatus(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	jobRunId, taskRunId, sourceId := getQueryParams(r)
+	if jobRunId == "" {
+		http.Error(w, "job_run_id not found", http.StatusBadRequest)
+		return
+	}
+
+	if err := h.service.DeleteJobStatus(
+		ctx,
+		jobRunId,
+		rsources.JobFilter{TaskRunID: taskRunId, SourceID: sourceId},
+	); err != nil {
+		httpStatus := http.StatusInternalServerError
+		switch {
+		case errors.Is(err, rsources.ErrStatusNotFound):
+			httpStatus = http.StatusNotFound
+		}
+		http.Error(w, err.Error(), httpStatus)
```

Comment:
Since we are introducing a new "v2" I wanted to bring this up.

HTTP error returns in plain text response. This might be ok, especially if the client uses the status code instead of reading the body. Usually, you want an API to consistently return JSON in all endpoints and status code.

Do we want this uniformity here? What is our contract with sources team.

No a blocking comment, we can revisit this until v2 starts being used.